shard.go

```diff
@@ -0,0 +1,133 @@
+package flatfs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type ShardIdV1 struct {
+	funName string
+	param   int
+	fun     ShardFunc
+}
+
+const PREFIX = "/repo/flatfs/shard/"
+
+func (f *ShardIdV1) String() string {
+	return fmt.Sprintf("%sv1/%s/%d", PREFIX, f.funName, f.param)
+}
+
+func (f *ShardIdV1) Func() ShardFunc {
+	return f.fun
+}
+
+func ParseShardFunc(str string) (*ShardIdV1, error) {
+	str = strings.TrimSpace(str)
+	// ignore prefix for now
+	if len(str) == 0 {
+		return nil, fmt.Errorf("empty shard identifier")
+	}
+	if str[0] == '/' {
+		if !strings.HasPrefix(str, PREFIX) {
+			return nil, fmt.Errorf("invalid prefix in shard identifier: %s", str)
+		}
+		str = str[len(PREFIX):]
+	}
+	parts := strings.Split(str, "/")
+	if len(parts) == 3 {
```

Comment:
I don't agree (with "it doesn't have to be") but @whyrusleeping would probably say the same thing so I will change it.